server/sonar-process/src/main/java/org/sonar/process/StopWatcher.java

```diff
@@ -45,23 +45,19 @@ public StopWatcher(ProcessCommands commands, Stoppable stoppable) {
 
   @Override
   public void run() {
-    commands.prepare();
-    try {
-      while (watching) {
-        if (commands.askedForStop()) {
-          LoggerFactory.getLogger(getClass()).info("Stopping process");
-          stoppable.stopAsync();
+    while (watching) {
+      commands.ping();
```

Comment:
Two options:
* rename the class as it does not do only stop watching but also pinging
* remove ping() as it's still not used for autokill
I'm for option 2.